openmc/mesh.py

```diff
@@ -799,6 +799,43 @@ def write_data_to_vtk(self, filename, datasets, volume_normalization=True):
             volume_normalization=volume_normalization
         )
 
+    def get_data_slice(self, dataset: np.array, basis: str, slice_index: int):
+        """Maps the provided dataset values to the mesh and obtains a 2D slice
+        of dataset values on the specified mesh slice index. Useful when plotting
+        slices of mesh tally results.
+
+        Parameters
+        ----------
+        datasets : numpy.array
+            1-D array of data values. Will be reshaped to fill the mesh and
+            must have the same number of entries to fill the mesh
+        basis : {'xy', 'xz', 'yz'}
+            The basis directions for the slice
+        slice_index : int
+            The index of the mesh slice to extract.
+
+        Returns
+        -------
+        np.array()
+            the 2D array of dataset values
+        """
+
+        reshaped_ds = dataset.reshape(self.dimension, order="F")
+
+        if basis == "yz":
+            transposed_ds = reshaped_ds.transpose(0, 1, 2)[slice_index]
```

Comment:
Isn't this really the same as just
```suggestion
            transposed_ds = reshaped_ds[slice_index]
```